src/modm/platform/core/e31/clock.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2019, Benjamin Weps
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/architecture/interface/clock.hpp>
+
+#define RTC_COUNT_LO *((volatile uint32_t *)0x10000048)
+#define RTC_COUNT_HI *((volatile uint32_t *)0x1000004C)
+
+modm::Clock::Type modm::Clock::time;
+
+template< typename TimestampType >
+TimestampType
+modm::Clock::now()
+{
+    uint64_t ctrVal = RTC_COUNT_HI;
+    ctrVal = ctrVal << 32;
+    ctrVal |= RTC_COUNT_LO;
```

Comment:
This isn't atomic, so the value can actually go backwards in time, if the LO overflows into HI.
Can you try a 64-bit read? `*((volatile uint64_t *)0x10000048)`. Perhaps the bus does this read atomically then.

Reply:
I can't find a way right now to do an atomic 64 bit read. However I'm also not happy with the RTC setup. I will change this to a timer based version, once the interrupts are working